File: 1768-merge-strings-alternately/1768-merge-strings-alternately.c


char * mergeAlternately(char * word1, char * word2){
    int len_1 = strlen(word1);
    int len_2 = strlen(word2);

    char* result = (char*)malloc((len_1 + len_2 + 1) *sizeof(char));

    int i = 0; 
    int j = 0; 
    int k = 0;

    while(i < len_1 && j < len_2){
        result[k++] = word1[i++];
        result[k++] = word2[j++];
    }

    while(i < len_1){
        result[k++] = word1[i++];
    }

    while(j < len_2){
        result[k++] = word2[j++];
    }

    result[k] = '\0';

    return result;
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        w1 = len(word1)
        w2 = len(word2)
        result = []
        i = 0
        j = 0

        while i < w1 or j < w2:
            if i < w1:
                result.append(word1[i])
                i += 1

            if j < w2:
                result.append(word2[j])
                j += 1
        return "".join(result)